eureka-client/src/main/java/com/netflix/discovery/shared/Application.java

```diff
@@ -216,14 +216,15 @@ private void _shuffleAndStoreInstances(boolean filterUpInstances, boolean indexB
         synchronized (instances) {
             instanceInfoList = new ArrayList<InstanceInfo>(instances);
         }
-        if (indexByRemoteRegions || filterUpInstances) {
+        boolean remoteIndexingActive = indexByRemoteRegions && null != instanceRegionChecker && null != clientConfig
```

Comment:
The behavior through here doesn't look quite equivalent anymore.  It seems the old behavior would have gotten non-UP instances filtered out even when an instanceRegionChecker was null for example, but that the new behavior won't end up filtering out non-UP instances?